### PR TITLE
Infrastructure bug fixes and reliability improvements

### PR DESCRIPTION
**Pull request by robertschaedler3**
Bug fixes/infrastructure reliability improvements in E2E/`moduletest`  workflows:

- Replace GitHub custom runners with 1ES managed pool/images
  - VMs provisioned by 1ES pool using managed images (removed dependency on `terraform` and `cloud-init` for provisioning custom hosts)
  - Fixes reliability issues related to base image updates/packages/dependencies
  - Remove dependency on Personal Access Tokens (PATs) for provisioning custom runners
- Remove intermediate use of Azure Key Vault for storing/retrieving "dynamic" secrets such as a device connection string 
- Remove reliance on deployment environments for workflow secrets (simplify secrets needed for E2E workflows)
- Update `terraform` scripts to prevent resource name collisions during concurrent E2E test runs
- Refactor E2E tests
  - Remove dependency on local configuration (fixes race condition on RC/DC files during test execution)
  - Remove skipped tests
  - Modify `osconfig.json` to only include reported objects for modules with E2E tests
  - Remove/uninstall modules that do not have E2E tests
- Reduce repeated recursive-clone/build steps (include `moduletest`/shared-object binaries in package artifacts)
- Add Debian 10/11 to test matrices
- Remove disabled/obsolete workflows
- Enable manual triggers for Module Test workflow

### Checklist

- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [x] I added unit-tests to validate my changes. All unit tests are passing.
- [x] I have merged the latest `main` branch prior to this PR submission.
- [x] I submitted this PR against the `main` branch.

### CODE REVIEW

File: src/tests/e2etest/CommandRunnerTests.cs [hunk 1]
```diff
@@ -124,43 +142,42 @@ public void RefreshCommandStatus(string newCommandId, bool exptectFailure = fals
...
-        [TestCase("echo 'single\nline'", 0, true, 0, "single line ", CommandState.Succeeded, DataSourceType.IotHub)]
-        [TestCase("echo 'multiple\nlines'", 0, false, 0, "multiple\nlines\n", CommandState.Succeeded, DataSourceType.IotHub)]
-        [TestCase("blah", 0, false, 127, "sh: 1: blah: not found\n", CommandState.Failed, DataSourceType.IotHub)]
-        public void CommandRunnerTest_RunCommand(string arguments, int timeout, bool singleLineTextResult, int resultCode, string textResult, CommandState state, DataSourceType dataSourceType)
+        [TestCase("echo 'hello world'", 0, false, 0, "hello world\n", CommandState.Succeeded)]
+        // [TestCase("sleep 10s", 1, true, 62, "", CommandState.TimedOut)]
+        // [TestCase("sleep 10s", 60, true, 0, "", CommandState.Succeeded)]
+        // [TestCase("echo 'single\nline'", 0, true, 0, "single line ", CommandState.Succeeded)]
+        // [TestCase("echo 'multiple\nlines'", 0, false, 0, "multiple\nlines\n", CommandState.Succeeded)]
+        // [TestCase("blah", 0, false, 127, "sh: 1: blah: not found\n", CommandState.Failed)]
```
Comment (MariusNi) [inline src/tests/e2etest/CommandRunnerTests.cs:160]: Remove?

File: src/tests/e2etest/DeliveryOptimizationTests.cs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the MIT License.
 
 using NUnit.Framework;
+using System.Threading.Tasks;
```
Comment (MariusNi) [inline src/tests/e2etest/DeliveryOptimizationTests.cs:5]: DeliveryOptimization is not a good module for E2E. Remove?

**Comment by MariusNi** [comment]
> using System.Text.RegularExpressions;

If Firewall is super reliable for E2E, keep this and remove all others

---
Refers to: src/tests/e2etest/FirewallTests.cs:5 in 7320179. [](commit_id = 73201791ba9762341e798e6bf2b271b9184eb480, deletion_comment = False)

**Comment by MariusNi** [comment]
> using System;

Remove Networking from E2E tests, this module is not right for this.

---
Refers to: src/tests/e2etest/NetworkingTests.cs:5 in 7320179. [](commit_id = 73201791ba9762341e798e6bf2b271b9184eb480, deletion_comment = False)

**Comment by MariusNi** [comment]
> namespace E2eTesting

Remove Tpm from E2E test, module is not right for this

---
Refers to: src/tests/e2etest/TpmTests.cs:8 in 7320179. [](commit_id = 73201791ba9762341e798e6bf2b271b9184eb480, deletion_comment = False)

File: src/tests/e2etest/E2ETest.cs
Comment (AhmedBM) [inline src/tests/e2etest/E2ETest.cs:1]: No more local RC validation? Looks like only the hub is present now?

File: .github/workflows/deploy-insiders-fast.yml
Comment (AhmedBM) [inline .github/workflows/deploy-insiders-fast.yml:1]: No longer testing pros/insider installs?

File: .github/workflows/e2e-run.yml [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+          reported=$(cat ./devops/e2e/reported.json)
+          config=$(sudo jq --argjson reported "$reported" '.ReportingIntervalSeconds = 5 | .Reported = $reported' /etc/osconfig/osconfig.json)
+          echo $config | sudo tee /etc/osconfig/osconfig.json
+
+          # Remove modules that do not have associated E2E tests
+          sudo rm /usr/lib/osconfig/configuration.so
+          sudo rm /usr/lib/osconfig/deviceinfo.so
+          sudo rm /usr/lib/osconfig/pmc.so
+          sudo rm /usr/lib/osconfig/securitybaseline.so
+          sudo rm /usr/lib/osconfig/ztsi.so
```
[the remark this answers is not in the document]
Comment (AhmedBM) [reply .github/workflows/e2e-run.yml:89]: Why remove any modules? Do we not want to catch any issues with the package as it ships?

File: src/tests/e2etest/E2ETest.cs
Comment (robertschaedler3) [reply src/tests/e2etest/E2ETest.cs:1]: Yes, local RC/DC validation had significant issues due to a race condition on the local files (especially when the document is large/contains many reported properties). As the RC file grew is size with more modules/reported properties added, the probability that the E2E tests would attempt to read the file before it was created increased (similar issue of reading/writing at the same time casuing sporatic failures).

File: .github/workflows/e2e-run.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (robertschaedler3) [reply .github/workflows/e2e-run.yml:89]: Modules are removed to isolate the tests to test the E2E behavior of osconfig via the IoTHub (detect potential issues in the PnP adapter/platform).

File: .github/workflows/deploy-insiders-fast.yml
Comment (robertschaedler3) [reply .github/workflows/deploy-insiders-fast.yml:1]: The package tested is the same. We consistently test package installs via E2E workflows. When insiders/prod releases occur that package has already been tested for each change that was checked in.

**Pull request merged**